### PR TITLE
ngx.worker.id return nil in other process like cache manger

### PR DESCRIPTION
**Pull request by cuiweixie**
Hi, I have heard that somebody may need this in google group

### CODE REVIEW

File: t/135-worker-id.t [hunk 1]
```diff
@@ -31,3 +31,36 @@ GET /lua
 --- no_error_log
 [error]
 --- skip_nginx: 3: <=1.9.0
+
```
Comment (agentzh) [inline t/135-worker-id.t]: Need to run the `reindex` script on this test file to fix formatting inconsistencies like this.

File: src/ngx_http_lua_worker.c [hunk 2]
```diff
@@ -93,6 +93,10 @@ int
 ngx_http_lua_ffi_worker_id(void)
 {
 #if (nginx_version >= 1009001)
+    if (ngx_process != NGX_PROCESS_WORKER) {
+        lua_pushnil(L);
+        return 1;
+    }
```
Comment (agentzh) [inline src/ngx_http_lua_worker.c]: Style: need a blank line after code blocks.

File: t/135-worker-id.t [hunk 3]
```diff
@@ -31,3 +31,36 @@ GET /lua
...
+=== TEST 2: worker id nil on no worker process
+--- http_config 
+         proxy_buffering on;
+         proxy_cache_valid any 10m;
+         proxy_cache_path /data1/cache levels=1:2 keys_zone=my-cache:8m max_size=1000m inactive=600m;
+         proxy_temp_path /data/temp;
+         proxy_buffer_size 4k;
+         proxy_buffers 100 8k;
+         init_worker_by_lua_block {
+         ngx.log(ngx.ERR, "worker id ",ngx.worker.id());
```
Comment (agentzh) [inline t/135-worker-id.t]: Style: need a space after `,`.

Comment (agentzh) [inline t/135-worker-id.t]: Also, need to fix indentation here.

**Comment by agentzh** [comment]
@cuiweixie thanks for taking care of this! Will you fix those small issues in my comments? Thanks!

**Comment by cuiweixie** [comment]
@agentzh  please check it again!

**Comment by agentzh** [comment]
@cuiweixie I think we should fix the `ngx_http_lua_ngx_worker_id` function too?

**Comment by agentzh** [comment]
@cuiweixie Also, please rebase to the latest git master. Thank you!

**Pull request closed without merging**